src/core/sys/windows/stacktrace.d

```diff
@@ -345,26 +345,38 @@ extern(Windows) BOOL FixupDebugHeader(HANDLE hProcess, ULONG ActionCode,
     return FALSE;
 }
 
-private string generateSearchPath()
+private char[] generateSearchPathAlloc()
 {
-    __gshared string[3] defaultPathList = ["_NT_SYMBOL_PATH",
-                                           "_NT_ALTERNATE_SYMBOL_PATH",
-                                           "SYSTEMROOT"];
+    __gshared string[3] defaultPathList = [
+        "_NT_SYMBOL_PATH", "_NT_ALTERNATE_SYMBOL_PATH", "SYSTEMROOT"
+    ];
 
-    string path;
-    char[2048] temp = void;
+    char[MAX_PATH] temp = void;
     DWORD len;
+    DWORD total;
+    foreach (e; defaultPathList)
+    {
+        if ((len = GetEnvironmentVariableA(e.ptr, temp.ptr, temp.length)) > 0)
+        {
+            total += len + 1;
+        }
+    }
 
-    foreach ( e; defaultPathList )
+    if(total > 0)
     {
-        if ( (len = GetEnvironmentVariableA( e.ptr, temp.ptr, temp.length )) > 0 )
+        auto buffer = cast(char[]) malloc(total + 1)[0 .. total + 1];
+        foreach (e; defaultPathList)
```

Comment:
It is wasteful to iterate defaultPathList twice. You can integrate this loop in the one above.